src/libraries/System.Security.Cryptography.Encoding/tests/PemEncodingFindTests.cs

```diff
@@ -0,0 +1,381 @@
+        }
+
+        [Fact]
+        public void Find_Success_Minimum()
+        {
+            string content = "-----BEGIN ----------END -----";
+            PemFields fields = AssertPemFound(content,
+                expectedLocation: 0..30,
+                expectedBase64: 16..16,
+                expectedLabel: 11..11);
+            Assert.Equal(0, fields.DecodedDataLength);
+        }
+
+        [Fact]
+        public void Find_Success_PrecedingContentAndWhitespaceBeforePreeb()
+        {
+            string content = "boop   -----BEGIN TEST-----\nZm9v\n-----END TEST-----";
+            AssertPemFound(content,
+                expectedLocation: 7..51,
+                expectedBase64: 28..32,
+                expectedLabel: 18..22);
+        }
+
+        [Fact]
+        public void Find_Success_TrailingWhitespaceAfterPosteb()
+        {
+            string content = "-----BEGIN TEST-----\nZm9v\n-----END TEST-----    ";
+            AssertPemFound(content,
+                expectedLocation: 0..44,
+                expectedBase64: 21..25,
+                expectedLabel: 11..15);
+        }
+
+        [Fact]
+        public void Find_Success_EmptyLabel()
+        {
+            string content = "-----BEGIN -----\nZm9v\n-----END -----";
+            AssertPemFound(content,
+                expectedLocation: 0..36,
+                expectedBase64: 17..21,
+                expectedLabel: 11..11);
+        }
+
+        [Fact]
+        public void Find_Success_EmptyContent_OneLine()
+        {
+            string content = "-----BEGIN EMPTY----------END EMPTY-----";
+            PemFields fields = AssertPemFound(content,
+                expectedLocation: 0..40,
+                expectedBase64: 21..21,
+                expectedLabel: 11..16);
+            Assert.Equal(0, fields.DecodedDataLength);
+        }
+
+        [Fact]
+        public void Find_Success_EmptyContent_ManyLinesOfWhitespace()
+        {
+            string content = "-----BEGIN EMPTY-----\n\t\n\t\n\t  \n-----END EMPTY-----";
+            PemFields fields = AssertPemFound(content,
+                expectedLocation: 0..49,
+                expectedBase64: 30..30,
+                expectedLabel: 11..16);
+            Assert.Equal(0, fields.DecodedDataLength);
+        }
+
+        [Theory]
+        [InlineData("CERTIFICATE")]
+        [InlineData("X509 CRL")]
+        [InlineData("PKCS7")]
+        [InlineData("PRIVATE KEY")]
+        [InlineData("RSA PRIVATE KEY")]
+        public void Find_Success_CommonLabels(string label)
+        {
+            string content = $"-----BEGIN {label}-----\nZm9v\n-----END {label}-----";
+            PemFields fields = FindPem(content);
+            Assert.Equal(label, content[fields.Label]);
+        }
+
+        [Theory]
+        [InlineData("H E L L O")]
+        [InlineData("H-E-L-L-O")]
+        [InlineData("HEL-LO")]
+        [InlineData("H")]
+        public void TryFind_True_LabelsWithHyphenSpace(string label)
```

Comment:
"H" has no hyphens or spaces, so ideally either it moves to a different method, or the name just gets updated here.

It also looks like the name says TryFind but the body uses Find.